cli/utils.js

```diff
@@ -47,7 +47,7 @@ const error = function(msg, trace) {
     if (trace) {
         console.log(trace);
     }
```

Comment:
By removing process.exit at this point, jhipster will not exit on log.error anymore.
We should find a cleaner way to exit the process.

Reply:
Maybe throwing an exception.

Reply:
@mshima: Thanks for caching this. Yes, throwing an exception seems to be the way that the documentation suggests; https://nodejs.org/api/process.html#process_process_exit_code and https://nodejs.dev/error-handling-in-nodejs. Let me try to correct this in a separate PR. Or would you like to do the PR? :smile:

Reply:
Feel free to do it. Don't have time right now.

Reply:
Btw, I've just tested this function and when we set `process.exitCode=1` and for me, it seems to exit cleanly without any problems. Could you give me and example where you see this not exiting cleanly problem? :thinking:

Reply:
I didn't say it's wrong, but is not safe for last minute merge just before release.

There is a change of behavior on logger.error (Exit on first error to set exit status and keep running)
I found 10 uses of logger.error, so we must check this new behavior.

https://github.com/jhipster/generator-jhipster/blob/08f36bbe77a92140516dc4a50401a202c9001f38/cli/jhipster.js#L28-L32

Currently, if node version is not supported, jhipster will log the error and quit.
With the new behavior, jhipster will keep running and maybe will success, but jhipster will exit with error code.

At this point, probably we should change to something like logger.fatal().

We must check done behavior too:
https://github.com/jhipster/generator-jhipster/blob/08f36bbe77a92140516dc4a50401a202c9001f38/cli/utils.js#L184-L190

Reply:
Ah, I understand now. We'll keep this as it is for now I think; since changing it might involve unnecessary problems. I've changed the other two and created #10956; feel free to let me know if you see any further issues. Thanks :smile: